### PR TITLE
Add KMS Key

### PR DESCRIPTION
**Pull request by tomasdembelli**
- Add KMS Key
- Add integration tests for KMS Key

### CODE REVIEW

File: sources/kms/key.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		Attributes:      attributes,
+		Scope:           scope,
+		Tags:            resourceTags,
+	}
+
+	switch output.KeyMetadata.KeyState {
+	case types.KeyStateEnabled:
+		item.Health = sdp.Health_HEALTH_OK.Enum()
+	case types.KeyStateDisabled:
+		item.Health = sdp.Health_HEALTH_ERROR.Enum()
```
Comment (dylanratcliffe) [inline sources/kms/key.go:57]:
```suggestion
		item.Health = sdp.Health_HEALTH_UNKNOWN.Enum()
```

I don't think disabled is necessarily bad, let's not mark it as unhealthy

File: sources/kms/key.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+	item := &sdp.Item{
+		Type:            "kms-key",
+		UniqueAttribute: "keyId",
+		Attributes:      attributes,
+		Scope:           scope,
+		Tags:            resourceTags,
+	}
+
+	switch output.KeyMetadata.KeyState {
```
Comment (dylanratcliffe) [inline sources/kms/key.go:53]: To address this lint issue I'd suggest we explicitly use call them all out i.e.

```go
	switch output.KeyMetadata.KeyState {
	case types.KeyStateEnabled:
		item.Health = sdp.Health_HEALTH_OK.Enum()
	case types.KeyStateDisabled:
		item.Health = sdp.Health_HEALTH_ERROR.Enum()
	case types.KeyStateUnavailable:
		item.Health = sdp.Health_HEALTH_UNKNOWN.Enum()
	case types.KeyStateCreating,
		types.KeyStatePendingDeletion,
		types.KeyStatePendingImport,
		types.KeyStatePendingReplicaDeletion,
		types.KeyStateUpdating:
		item.Health = sdp.Health_HEALTH_PENDING.Enum()
	}
```

That way if they add a new thing, the linter will catch it for us

File: docs-data/kms-key.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (dylanratcliffe) [inline docs-data/kms-key.json:1]: Can you rerun `go generate ./...` to get this updated?

File: go.mod [hunk 4]
```diff
@@ -115,3 +115,5 @@ require (
 	gopkg.in/ini.v1 v1.67.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
+
+require github.com/aws/aws-sdk-go-v2/service/kms v1.35.3
```
Comment (dylanratcliffe) [inline go.mod:119]: Can you move this into the upper block with the other direct dependencies?

**Review by dylanratcliffe — CHANGES_REQUESTED** [review]
A few notes

File: sources/kms/key.go (hunk 2 above)
Comment (tomasdembelli) [reply sources/kms/key.go:53]: Totally agree, this was my initial thought as well.

I've added a `default` case that returns error which would give us a warning.

File: sources/kms/key_test.go [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+	"github.com/overmindtech/aws-source/sources"
+
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+)
+
+type testClient struct{}
+
+func (t testClient) DescribeKey(ctx context.Context, params *kms.DescribeKeyInput, optFns ...func(*kms.Options)) (*kms.DescribeKeyOutput, error) {
+	return &kms.DescribeKeyOutput{
+		KeyMetadata: &types.KeyMetadata{
```
[the remark this answers is not in the document]
Comment (tomasdembelli) [reply sources/kms/key_test.go:18]: Agree, I was going to come back to it when working on custom key store, but you are right, better to put it first and fix it if necessary.

[the remark this answers is not in the document]
Comment (dylanratcliffe) [reply sources/kms/key_test.go:18]: Yeah that's my usual approach. As long as you follow the [naming convention](https://github.com/overmindtech/aws-source?tab=readme-ov-file#naming-conventions) you can be pretty sure that it's going to be okay. For example if you search there are already loads of references to `kms-key`

**Review by dylanratcliffe — APPROVED** [review]

**Pull request merged**